PythonPractice/TicTacToe/tictactoe.py

```diff
@@ -0,0 +1,155 @@
+class SpaceTakenError(Exception):
+    pass
+
+class BoardTooSmallError(Exception):
+    pass
+
+class TicTacToe:
+
+    def __init__(self):
+        self.board = []
+        self.initialize_board()
+        self.playerO = 'O'
+        self.playerX = 'X'
+        self.player = self.playerX
+
+    def initialize_board(self):
+        """
+        Creates an empty board to be played on
+        """
+        # get board size from user
+        good_input = False
+        size = None
+        while not good_input:
+            try:
+                size = int(raw_input("What size board to do you want to play on?\n"))
+                if size < 3:
+                    raise BoardTooSmallError
+            except BoardTooSmallError:
+                print "Your board must be at least a 3x3 to play."
+            except ValueError:
+                print "Your board size must be an integer."
+            else:
+                good_input = True
+        # initialize board to empty strings
+        self.board = [[None for row in range(0, size)] for col in range(0, size)]
+        self.size = len(self.board)
+        self.num_turns = size*size
+        self.draw_board()
+
+    def draw_board(self):
+        """
+        Draws current state of the board
+        """
+        for row in self.board:
+            print "---" * self.size
+            for col in row:
+                if col:
+                    print "|%s" % col,
+                else:
+                    print "| ",
+            print "|"
+        print "---" * self.size
+
+    def play(self):
+        """
+        If there are turns left, allows a play
+        Makes sure the user input for row and column is valid (positive integer, untake space, in bounds)
+        Sets the space according to the player's turn
+        Switches player before exiting
+        Output:
+            True if there are remaining turns
+            False if there are no remaining turns
+        """
+        if self.num_turns > 0:
+            good_input = False
+            row = None
+            col = None
```

Comment:
You don't need to initialize row and col here, as there's no case where they'd be referenced before they get assigned by the two raw_input calls further down.